File: Objective-C Version/GooeySlideMenuDemo/GooeySlideMenuDemo/Classes/SlideMenuButton.m
//
//  SlideMenuButton.m
//  GooeySlideMenuDemo
//
//  Created by Kitten Yang on 15/8/13.
//  Copyright (c) 2015年 Kitten Yang. All rights reserved.
//

#import "SlideMenuButton.h"

@interface SlideMenuButton()

@property(nonatomic,strong)NSString *buttonTitle;

@end

@implementation SlideMenuButton


-(id)initWithTitle:(NSString *)title{
    
    self = [super init];
    if (self) {
        
        self.buttonTitle = title;
        
    }
    return self;
    
}

// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {

    CGContextRef context = UIGraphicsGetCurrentContext();
    CGContextAddRect(context, rect);
    [self.buttonColor set];
    CGContextFillPath(context);

    
    UIBezierPath* roundedRectanglePath = [UIBezierPath bezierPathWithRoundedRect:CGRectInset(rect, 1, 1) cornerRadius: rect.size.height/2];
    [self.buttonColor setFill];
    [roundedRectanglePath fill];
    [[UIColor whiteColor] setStroke];
    roundedRectanglePath.lineWidth = 1;
    [roundedRectanglePath stroke];
    
    NSMutableParagraphStyle *paragraphStyle = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
    paragraphStyle.alignment = NSTextAlignmentCenter;
    
    
    
    NSDictionary *attr = @{NSParagraphStyleAttributeName:paragraphStyle,NSFontAttributeName:[UIFont systemFontOfSize:24.0f],NSForegroundColorAttributeName:[UIColor whiteColor]};
    CGSize size = [self.buttonTitle sizeWithAttributes:attr];
    
    CGRect r = CGRectMake(rect.origin.x,
                          rect.origin.y + (rect.size.height - size.height)/2.0,
                          rect.size.width,
                          size.height);

    [self.buttonTitle drawInRect:r withAttributes:attr];
    
}


- (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event {
    UITouch *touch = [touches anyObject];
    NSUInteger tapCount = touch.tapCount;
    
    
    switch (tapCount) {
        case 1:
            self.buttonClickBlock();
            break;
    
        default:
            break;
    }
    
}





@end

File: Swift Version/GooeySlideMenuDemo-Swift/GooeySlideMenuDemo-Swift/Classes/SlideMenuButton.swift
//
//  SlideMenuButton.swift
//  GooeySlideMenuDemo-Swift
//
//  Created by Kitten Yang on 1/3/16.
//  Copyright © 2016 Kitten Yang. All rights reserved.
//

import UIKit

struct MenuButtonOptions {
    var title: String
    var buttonColor: UIColor
    var buttonClickBlock: ()->()
}

class SlideMenuButton: UIView {

    private var _option: MenuButtonOptions
    
    init(option: MenuButtonOptions) {
        _option = option
        super.init(frame: CGRectZero)
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func drawRect(rect: CGRect) {
        let context = UIGraphicsGetCurrentContext()
        CGContextAddRect(context, rect)
        _option.buttonColor.set()
        CGContextFillPath(context)
        
        let roundedRectanglePath = UIBezierPath(roundedRect: CGRectInset(rect, 1, 1), cornerRadius: rect.height / 2)
        _option.buttonColor.setFill()
        roundedRectanglePath.fill()
        UIColor.whiteColor().setStroke()
        roundedRectanglePath.lineWidth = 1
        roundedRectanglePath.stroke()

        let paragraphStyle = NSParagraphStyle.defaultParagraphStyle().mutableCopy() as! NSMutableParagraphStyle
        paragraphStyle.alignment = .Center
        let attr = [NSParagraphStyleAttributeName: paragraphStyle, NSFontAttributeName: UIFont.systemFontOfSize(24.0),NSForegroundColorAttributeName: UIColor.whiteColor()]
        let size = _option.title.sizeWithAttributes(attr)
        
        let r = CGRectMake(rect.origin.x,
            rect.origin.y + (rect.size.height - size.height)/2.0,
            rect.size.width,
            size.height)
        _option.title.drawInRect(r, withAttributes: attr)

    }
    
    override func touchesEnded(touches: Set<UITouch>, withEvent event: UIEvent?) {
        let touch = touches.first
        if let touch = touch {
            let tapCount = touch.tapCount
            switch (tapCount) {
                case 1: _option.buttonClickBlock()
                default: break
            }
        }
    }
}
